source/beamline/SGM/SGMHexapodTransformedAxis.cpp

```diff
@@ -185,6 +200,111 @@ AMAction3 * SGMHexapodTransformedAxis::createSetParameterActions(double startPoi
 	return action;
 }
 
+AMAction3 *SGMHexapodTransformedAxis::createSetParametersActions(const QVector3D &startPoint, const QVector3D &endPoint, double deltaTime)
+{
+	AMListAction3* action = 0;
+	lastDeltaTime_ = deltaTime;
+
+	qDebug() << "Checking on some axis stuff for SGMHexapodTransformedAxis::createSetParameterActions for axis " << axis_;
+	if(globalXAxis_)
+		qDebug() << "Have globalXAxis_";
+	if(globalYAxis_)
+		qDebug() << "Have globalYAxis_";
+	if(globalZAxis_)
+		qDebug() << "Have globalZAxis_";
+	if(trajectoryStartControl_)
+		qDebug() << "Have trajectoryStartControl_";
+	if(globalXAxis_ && globalYAxis_ && globalZAxis_ && trajectoryStartControl_) {
+
+		// Setting the start position
+
+		action = new AMListAction3(new AMListActionInfo3("Setting hexapod movement parameters",
+								 "Setting hexapod movement parameters"),
+					   AMListAction3::Sequential);
+
+
+
+		// Create the required move actions in the global system:
+		AMListAction3* startMoveActions = new AMListAction3(new AMListActionInfo3("Moving parameter controls for start",
+										     "Moving parameter controls for end"),
+													   AMListAction3::Parallel);
+
+		startMoveActions->addSubAction(AMActionSupport::buildControlMoveAction(globalXAxis_, startPoint.x()));
```

Comment:
It looks, from usage, that this function is passed vectors which are in the prime coordinate system, and is then using them to set controls for the global system. I think this needs a primeToGlobal() call in here before they are used.